### PR TITLE
Add Directory support for S3 PutObject + Debug Logging Support

### PR DESCRIPTION
**Pull request by bradyoo**
When the Key ends with a separator, it means a directory should be added

### CODE REVIEW

File: core/server/proxy/src/main/java/alluxio/proxy/s3/logging/RequestLoggingFilter.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Helper class for adding logs to request methods for debugging purposes.
+ */
+@Provider
+public class RequestLoggingFilter implements ContainerRequestFilter {
+  private static final Logger LOG = LoggerFactory.getLogger(RequestLoggingFilter.class);
+
+  @Override
+  public void filter(ContainerRequestContext context) throws IOException {
+    LOG.info("Method: " + context.getMethod());
```
Comment (gpang) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/logging/RequestLoggingFilter.java:31]: Do these need to be info level?

**Review by gpang — COMMENTED** [review]
@bradyoo just a minor question

File: core/server/proxy/src/main/java/alluxio/proxy/s3/logging/RequestLoggingFilter.java (hunk 1 above)
Comment (bradyoo) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/logging/RequestLoggingFilter.java:31]: This code is not actually included in any of our codebase by default. You have to add it with Logged annotation. So in that case, info is just more convenient.

**Review by gpang — APPROVED** [review]
LGTM

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**